lib/pf/SwitchFactory.pm

```diff
@@ -165,6 +179,36 @@ sub config {
     return \%temp;
 }
 
+sub getModule {
+    my ($type) = @_;
+    return (exists $TYPE_TO_MODULE{$type} ) ? $TYPE_TO_MODULE{$type} : undef;;
+}
+
+=item buildVendorsList
+
+Build the vendor list
+
+Example Usage:
+
+    buildVendorsList()
+
+Return: N/A
+
+=cut
+
+sub buildVendorsList {
+    for my $module (@MODULES) {
+        my $switch = $module;
+        $switch =~ s/^pf::Switch:://;
+        my @p = split /::/, $switch;
+        my $vendor = shift @p;
+        if ($module->can('description')) {
```

Comment:
I'm not saying it doesn't accomplish the goal, it's more a question of 'is it clean ?'